flexget/plugins/output/decompress.py

```diff
@@ -25,53 +25,60 @@ def open_archive_entry(entry):
     """
     Convenience method for opening archives from entries. Returns an archive.Archive object
     """
-    arch = None
+    archive = None
 
     try:
         archive_path = entry['location']
-        arch = archive.open_archive(archive_path)
-    except KeyError:
+        if not archive_path:
+            raise ValueError()
+
+        archive = archiveutil.open_archive(archive_path)
+    except (KeyError, ValueError):
         log.error('Entry does not appear to represent a local file.')
-    except archive.BadArchive as error:
+    except archiveutil.BadArchive as error:
         fail_entry_with_error(entry, 'Bad archive: %s (%s)' % (archive_path, error))
-    except archive.NeedFirstVolume:
+    except archiveutil.NeedFirstVolume:
         log.error('Not the first volume: %s', archive_path)
-    except archive.ArchiveError as error:
+    except archiveutil.ArchiveError as error:
         fail_entry_with_error(entry, 'Failed to open Archive: %s (%s)' % (archive_path, error))
 
-    return arch
+    return archive
 
 
-def get_destination_path(path, to, keep_dirs):
-    """
-    Generate the destination path for a given file
-    """
-    filename = os.path.basename(path)
-
-    if keep_dirs:
-        path_suffix = path
+def get_output_path(to, entry):
+    """Determine which path to output to"""
+    if to:
+        return render_from_entry(to, entry)
     else:
-        path_suffix = filename
+        return os.path.dirname(entry.get('location'))
 
-    return os.path.join(to, path_suffix)
 
+def extract_info(info, archive, to, keep_dirs):
+    """Extract ArchiveInfo object"""
 
-def is_dir(info):
-    """
-    Tests whether the file descibed in info is a directory
-    """
+    destination = get_destination_path(info, to, keep_dirs)
+
+    log.debug('Attempting to extract: %s to %s', info.filename, destination)
+    try:
+        info.extract(archive, destination)
+    except archiveutil.FSError as error:
+        log.error('OS error while creating file: %s (%s)' % (destination, error))
+    except archiveutil.ArchiveError as error:
+        log.error('Failed to extract file: %s in %s (%s)' % (info.filename,
+                                                                   entry['location'], error))
+    except archiveutil.FileAlreadyExists as error:
+        log.warn('File already exists: %s' % destination)
 
-    if hasattr(info, 'isdir'):
-        return info.isdir()
-    else:
-        base = os.path.basename(info.filename)
-        return not base
 
+def get_destination_path(info, to, keep_dirs):
+    """Generate the destination path for a given file"""
 
-def makepath(path):
-    if not os.path.exists(path):
-        log.debug('Creating path: %s', path)
-        os.makedirs(path)
+    if keep_dirs:
```

Comment:
style suggestion:
```python
path_suffix = info.path if keep_dirs else os.path.basename(info.path)
```

Reply:
It's not. I find it more readable in this case though, but I don't care either way.

Reply:
That's two votes in favor of it, so I'll do it. :)